python/ray/data/_internal/planner/plan_udf_map_op.py

```diff
@@ -280,17 +305,182 @@ def plan_udf_map_op(
     )
 
 
-def _get_udf(op: AbstractUDFMap):
+def plan_download_op(
+    op: Download,
+    physical_children: List[PhysicalOperator],
+    data_context: DataContext,
+) -> MapOperator:
+    assert len(physical_children) == 1
+    input_physical_dag = physical_children[0]
+    compute = get_compute(op._compute)
+    uri_column_name = op.uri_column_name
+
+    fn, init_fn = _get_udf(PartitionActor, (), {}, (uri_column_name,), {})
+    partition_transform_fn = _generate_transform_fn_for_map_batches(fn)
+    partition_transform_fns = [
+        # Convert input blocks to batches.
+        BlocksToBatchesMapTransformFn(
+            batch_size=None,  # TODO: Should we use op._batch_size here?
+            batch_format="pyarrow",
+            zero_copy_batch=True,
+        ),
+        # Apply the PartitionActor function.
+        BatchMapTransformFn(partition_transform_fn),
+    ]
+    partition_map_transformer = MapTransformer(partition_transform_fns, init_fn)
+    partition_map_operator = MapOperator.create(
+        partition_map_transformer,
+        input_physical_dag,
+        data_context,
+        name="DownloadURIPartitioner",
+        compute_strategy=compute,
+        ray_remote_args=op._ray_remote_args,
+        ray_remote_args_fn=op._ray_remote_args_fn,
+    )
+
+    fn, init_fn = _get_udf(download_images_threaded, (uri_column_name,), {}, (), {})
+    download_transform_fn = _generate_transform_fn_for_map_batches(fn)
+    transform_fns = [
+        # Apply the download function.
+        BatchMapTransformFn(download_transform_fn),
```

Comment:
Same here. Should this be `BlockMapTransformFn`? Inputs to a `MapOperator` guaranteed to blocks, not batches